proto/redpanda/api/dataplane/v1alpha2/secret.proto

```diff
@@ -49,8 +58,149 @@ message ListSecretsFilter {
       string: {pattern: "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"}
     }
   }];
+
+  // Secret scopes to search for
+  repeated Scope scopes = 3;
+}
+
+// Generic Secrets API
+
+// ListSecretsRequest is the request of ListSecrets.
+message ListSecretsRequest {
+  // List filter.
+  ListSecretsFilter filter = 1;
+
+  // Value of the next_page_token field returned by the previous response.
+  // If not provided, the system assumes the first page is requested.
+  string page_token = 2;
+
+  // Limit the paginated response to a number of items.
+  int32 page_size = 3 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+    description: "Limit the paginated response to a number of items."
+    minimum: 1
+    maximum: 50
+  }];
+}
+
+// GetSecretRequest is the request of GetSecret.
+message GetSecretRequest {
+  // The id of the secret to retrieve.
+  string id = 1 [
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 255,
+    (buf.validate.field).string.pattern = "^[A-Z][A-Z0-9_]*$"
+  ];
 }
 
+// GetSecretResponse is the response of GetSecret.
+message GetSecretResponse {
+  // The created secret.
+  Secret secret = 1;
+}
+
+// CreateSecretRequest is the request of CreateSecret.
+message CreateSecretRequest {
+  // Secret identifier.
+  string id = 1 [
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 255,
+    (buf.validate.field).string.pattern = "^[A-Z][A-Z0-9_]*$"
+  ];
+
+  // Secret labels.
+  map<string, string> labels = 2 [
+    (buf.validate.field).map = {
+      values: {
+        string: {pattern: "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"}
+      }
+    },
+    (google.api.field_behavior) = IMMUTABLE
+  ];
+
+  // Secret scopes
+  repeated Scope scopes = 3 [(buf.validate.field).repeated = {
+    min_items: 1
+    unique: true
+    items: {
+      enum: {
+        defined_only: true
+        not_in: [0]
+      }
+    }
+  }];
+
+  // The secret data.
+  bytes secret_data = 4 [
+    (google.api.field_behavior) = INPUT_ONLY,
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).bytes.max_len = 65536
+  ];
+}
+
+// CreateSecretResponse is the response of CreateSecret.
+message CreateSecretResponse {
+  // The created secret.
+  Secret secret = 1;
+}
+
+// UpdateSecretRequest is the request of UpdateSecret.
+message UpdateSecretRequest {
+  // Secret identifier.
+  string id = 1 [
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 255,
+    (buf.validate.field).string.pattern = "^[A-Z][A-Z0-9_]*$"
```

Comment:
I do not understand why this doesn't match the connect secrets pattern allowing lowercase?
Also isn't the first character set `[A-Z]` redundant since it's in `[A-Z0-9_]` ?

Reply:
One way we may want to fall back loading secrets in RPCN is via env-like style injection, see: https://docs.redpanda.com/redpanda-connect/configuration/secrets/ (cc @Jeffail )

The pattern indeed forces the secret to have an env var shape, i.e. start with letter and continue with alphanum or underscore.

We can consider this a starting point that makes things easier for us, since the secret name maps to how it's looked up in code (`${MY_SECRET_01}`). If this is too restrictive, we can later relax the pattern to allow other characters.